### PR TITLE
rpc test refactors

### PR DESCRIPTION
**Pull request by rgrinberg**
Just some simple changes to make the code more readable and minimize the diff in subsequent PR's.

### CODE REVIEW

File: test/expect-tests/dune_rpc_e2e/dune_rpc_e2e.ml [hunk 1]
```diff
@@ -180,4 +180,5 @@ let run run =
     ~finally:(fun () -> Sys.chdir cwd)
     ~f:(fun () ->
       Sys.chdir (Path.to_string dir);
-      Scheduler.Run.go config run ~timeout:5.0 ~on_event:(fun _ _ -> ()))
+      Scheduler.Run.go ~file_watcher:Scheduler.Run.No_watcher config run
```
Comment (snowleopard) [inline test/expect-tests/dune_rpc_e2e/dune_rpc_e2e.ml:183]: Shall we make the `file_watcher` mandatory? Btw, I'm guessing the compiler should be fine with just `~file_watcher:No_watcher` here (i.e. without the full module qualification).

**Review by snowleopard — APPROVED** [review]
Looks good. By the way, does "e2e" mean "exchange to exchange"? Never came across this abbreviation before, had to google. Maybe worth adding a comment on top of the corresponding mli.

File: test/expect-tests/dune_rpc_e2e/dune_rpc_e2e.ml (hunk 1 above)
Comment (rgrinberg) [reply test/expect-tests/dune_rpc_e2e/dune_rpc_e2e.ml:183]: yeah, it's worth considering. On the other hand, we can also factor the scheduler a little more to decouple it from the file watcher completely. Then one would need to always explicitly request file watching - which seems like the natural thing to do.

**Comment by rgrinberg** [comment]
"e2e" in this context means "end to end". So a little like our expect tests, but a bit too complex to write in bash.

File: test/expect-tests/dune_rpc_e2e/dune_rpc_e2e.ml (hunk 1 above)
Comment (snowleopard) [reply test/expect-tests/dune_rpc_e2e/dune_rpc_e2e.ml:183]: Yeah, that would be nice.

**Pull request closed without merging**